client/lib/domains/request-gdpr-consent-management-link.js

```diff
@@ -1,5 +1,5 @@
 import wpcom from 'calypso/lib/wp';
 
-export function requestGdprConsentManagementLink( domainName, onComplete ) {
-	return wpcom.undocumented().requestGdprConsentManagementLink( domainName, onComplete );
+export function requestGdprConsentManagementLink( domainName ) {
+	return wpcom.req.get( `/domains/${ domainName }/request-gdpr-consent-management-link` );
```

Comment:
Should we use `encodeURIComponent( domainName )` here? The `domainName` is a string, not a number. Can a valid domain name ever contain characters that need to be encoded?

Also, the `requestGdprConsentManagementLink` function could be inlined into the one place that calls it. A separate module is not needed.

Reply:
> Can a valid domain name ever contain characters that need to be encoded?

Hostnames are limited to a-z, 0-9, "-", and "."  So I think it's ok without encoding it here.

Reply:
Thanks for the confirmation @delputnam!

Reply:
There are several places in the codebase where we do `encodeURIComponent( domain )` or `encodeURIComponent( domainName )`, so I wasn't sure. Now we know we can remove these.